voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/AlexaInput.java

```diff
@@ -20,77 +20,81 @@
+import com.amazon.speech.speechlet.SpeechletRequest;
 
 class AlexaInput implements VoiceInput {
-  private IntentRequest request;
-  private Map<String,Object> metadata;
-  
+  private SpeechletRequest request;
+  private Map<String, Object> metadata = Collections.emptyMap();
+
   public AlexaInput(Object object) {
     this(object, null);
   }
 
-  @SuppressWarnings("unchecked")
-  public AlexaInput(Object request, Object metadata) {
-    if (!(request instanceof IntentRequest)) {
-      throw new IllegalArgumentException("First argument is not an instance of IntentRequest: " + request);
+  public AlexaInput(Object object, Map<String, Object> metadata) {
+    if (!(object instanceof SpeechletRequest)) {
+      throw new IllegalArgumentException("Argument is not an instance of SpeechletRequest: " + object);
     }
 
-    this.request = (IntentRequest) request;
-    
-    if(metadata != null){
-      if (!(metadata instanceof Map<?,?>)) {
-        throw new IllegalArgumentException("Second argument is not an instance of Map<?,?>: " + metadata);
-      }
-      this.metadata = (Map<String, Object>) metadata;
+    request = (SpeechletRequest) object;
+
+    if (metadata != null) {
+      this.metadata = metadata;
     }
   }
 
   @Override
-  public String getMessage() {
-    if (request.getIntent().getSlots() == null) {
-      return request.getIntent().getName();
-    }
-
-    StringBuilder buffer = new StringBuilder(request.getIntent().getName());
-    for (Entry<String, Slot> entry : request.getIntent().getSlots().entrySet()) {
-      buffer.append(' ');
-      buffer.append(entry.getValue().getValue());
+  public String getMessageAsString() {
+    if (!(request instanceof IntentRequest)) {
+      return "";
     }
 
-    return buffer.toString();
+    IntentRequest intentRequest = (IntentRequest) request;
+    return intentRequest.getIntent().getName();
   }
 
   @Override
-  public Map<String, String> getParameters() {
-    Map<String,String> slots = new HashMap<String,String>();
-    Intent intent = request.getIntent();
-    if(intent.getSlots() == null){
-      return slots;
+  public Map<String, String> getMessageAsMap() {
+    if (!(request instanceof IntentRequest)) {
+      return Collections.emptyMap();
     }
-    
-    for(Entry<String,Slot> entry : intent.getSlots().entrySet()){
-      slots.put(entry.getKey(), entry.getValue().getValue());
+
+    IntentRequest intentRequest = (IntentRequest) request;
+    Map<String, String> result = new LinkedHashMap<String, String>();
+    for (Entry<String, Slot> entry : intentRequest.getIntent().getSlots().entrySet()) {
+      Slot slot = entry.getValue();
+      if (slot != null) {
+        result.put(slot.getName(), slot.getValue());
+      }
     }
-    
-    return slots;
+
+    return result;
   }
 
-  //TODO: We should put some code in here to map built in AMAZON intents to other request types
   @Override
-  public <E extends Enum<E>> E getRequestName(Class<E> enumClass) {
```

Comment:
Also, you're already mixing things into a single enum - start conversation comes from getRequest().getType(), whereas help would be an intent retrieved via getRequest().getIntent().getName(), but both are represented in the MessageType enum.  With that paradigm, we have three options:

1. Decide everything that is a shared message type, and manually map to it
2. Remove HELP from that enum, and treat it as something that will be handled by the secondary switch
3.  Replace the MessageType enum with a generic one that can switch across all of them.